### DISCUSSION TITLE
SPR with React or Angular

### DISCUSSION THREAD

**Discussion by artecoop** [vercel/vercel#4291] [General]
Hello! 
I'm a now user and very happy with it.

I stumbled upon this link
https://vercel.com/blog/serverless-pre-rendering
and now I'm here asking if it is possible to use with bare react (or angular app), where apis will target a remote server.

I'm trying to understand if this could be a suitable option instead of prerender services like prerender.io

**Comment by wiiiimm** [comment]
Hi @artecoop, yes, SPR uses our smart CDN's edge caching feature to pre-render all sorts of serverless responses. I will be providing a more in-depth post in a bit to cover this in details.

**Reply by LearnWebCode** [reply]
Hi @williamli I'm also interested in using a serverless function to generate React/HTML serverside. I'm looking forward to your post about this. In the meantime, if I wanted to use the response from a serverless function as the HTML for a top level route in my Now site, is this the standard / idiomatic way of doing that?

This would be in my `now.json` config file:

```
{
  "rewrites": [{ "source": "/about", "destination": "/api/about" }]
}
```

This works, but I'm not sure if there's a more obvious / optimal way I should be setting this up.

**Comment by artecoop** [comment]
@williamli Thank you, I'll wait for your post!!!

thanks again

**Comment by artecoop** [comment]
There is any update on this? There is little to no further documentation on this

**Comment by emish89** [comment]
hi @williamli some news on it? thanks
